roles/ansible-test-splitter/files/list_changed_targets.py

```diff
@@ -412,14 +427,64 @@ def build_result_struct(self, batches):
         return result
 
 
+def read_pullrequest_body(project_name, pull_request):
+    change_url = "https://api.github.com/repos/%s/pulls/%d" % (
+        project_name,
+        pull_request,
+    )
+    return [x for x in requests.get(change_url).json().get("body").split("\n") if x]
+
+
+ZUUL_EXTRA_TARGETS = "zuul-extra-targets"
+ZUUL_TARGETS = "zuul-targets"
+ZUUL_RELEASES = "zuul-releases"
+
+
+def read_pullrequest_zuul_override(project_name, pull_request):
+
+    desc = read_pullrequest_body(project_name, pull_request)
+
+    def _extract_data(line):
+        data = (":".join(line.split(":")[1:])).replace(",", " ")
+        return [x for x in data.split() if x]
+
+    result = {}
+    for line in desc:
+        for key in (ZUUL_EXTRA_TARGETS, ZUUL_TARGETS, ZUUL_RELEASES):
+            if line.lower().startswith(key + ":"):
+                result[key] = _extract_data(line)
+
+    return result
+
+
 if __name__ == "__main__":
     args = parse_args(sys.argv[1:])
 
     collections = [Collection(i) for i in args.collection_to_tests]
     collections_names = [c.collection_name() for c in collections]
 
+    ansible_releases = args.ansible_releases
+    zuul_targets, zuul_extra_targets = [], []
+    pr_request = None
+    if args.project_name is not None and args.pull_request is not None:
+        pr_request = read_pullrequest_zuul_override(
+            args.project_name, args.pull_request
+        )
+
+    if pr_request:
+        release_to_test = pr_request.get(ZUUL_RELEASES)
+        if release_to_test:
+            tmp_releases = [rel for rel in release_to_test if rel in ansible_releases]
+            ansible_releases = tmp_releases
```

Comment:
I'm not sure I understand the use case behind this. What's the case where we'd want to limit the releases tested against on a single PR? This seems like the sort of thing that should be addressed in zuul config, not as a one-off on a PR. If a PR is failing for a specific release and we just ignore that release, then it's still going to be failing for every other PR.

Reply:
The scope of this change is debugging only, we are not supposed to merge any PR using these keywords
In case you want to limit the number of ansible releases to be tested for any reason, you can use this

Reply:
This bypass the `override-checkout` feature that is already in place. If you want to test a PR against a different branch, you can still prepare a quick PR in ansible-zuul-jobs that adjust the override-checkout, and Depends-On it. I don't think we need something like that in this script.

Reply:
all the keywords here are bypassing the `override-checkout` here so this PR makes no sense, it is a burden pain to create what you call a 'quick PR' each time you need to perform a particular test
This PR is not really for my debug, if you think it is not useful I will close it, but for me I found the utility of this